src/app/services/resolvers/repo-details-resolver.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, RouterStateSnapshot } from '@angular/router';
+import { forkJoin, Observable, of } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { Repository } from 'src/@types';
+import { CatalogueService } from '../catalogue/catalogue.service';
+
+export type RepoDetailsData = {
+  repo: Repository;
+  readme: string
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RepoDetailsResolverService implements Resolve<RepoDetailsData> {
+
+  constructor( private svc: CatalogueService ) { }
```

Comment:
svc stands for?

use `private readonly` to make this service `constant`

Moreover, I would like to call it catalogueService instead of svc